src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTokenExtensions.cs

```diff
@@ -168,6 +170,11 @@ public static bool IsBeginningOfStatementContext(this SyntaxToken token)
 
                 case SyntaxKind.ElseKeyword:
                     return true;
+
+                case SyntaxKind.AwaitKeyword:
+                    // await foreach
+                    // await using
+                    return true;
```

Comment:
this doesn't seem right.   "IsBeginningOfStatementContext" is more about where any statement can go.  Not all statements can go after 'await'.  i.e. you can't say `await return x;`.  is thsi really the right thing to do?  can we add tests for this as well?

Reply:
I see what you're saying. I've moved the logic to be specific to those two recommenders (using and foreach). Thanks